### ISSUE TITLE
Documentation on extra context

### ISSUE THREAD

**Issue by karantan** [audreyr/cookiecutter#863]
* Cookiecutter version: 1.4
* Template project url: /
* Python version: 3.5.1
* Operating System: OSX

### Description:
I am trying to inject extra context via `cookiecutter()` function.
### What I've run:
This is my code:
```
...
authentication_random = binascii.hexlify(os.urandom(20)).decode("utf-8")
authomatic_random = binascii.hexlify(os.urandom(20)).decode("utf-8")
session_random = binascii.hexlify(os.urandom(20)).decode("utf-8")

cookiecutter(
    'https://github.com/...',
    extra_context={
        'authentication_random': authentication_random,
        'authomatic_random': authomatic_random,
        'session_random': session_random,
    }
)
...
```
The above code actually works, the only problem is that I also had to put `authentication_random`, `authomatic_random` and `session_random` to cookiecutter.json like this:

```
...
    "authentication_random": "{{cookiecutter.authentication_random}}",
    "authomatic_random": "{{cookiecutter.authomatic_random}}",
    "session_random": "{{cookiecutter.session_random}}"
...
```

I'm guessing this works as intended. The only problem I see here is that this last part isn't documented. refs:

- http://cookiecutter.readthedocs.io/en/latest/advanced/injecting_content.html?highlight=extra%20context
- http://cookiecutter.readthedocs.io/en/latest/advanced/suppressing_prompts.html#advanced-example-defaults-extra-context

There is no mentioning that you also need to add these keys to cookiecutter.json file. IMO this should be added.

I can do this, just give me the green light.

**Comment by hackebrot** [comment]
Hi @karantan! 👋 

Thanks for raising this issue and offering to resolve it.

The [extra-context docs](http://cookiecutter.readthedocs.io/en/latest/advanced/injecting_content.html) mention that ``extra_context`` overrides cookiecutter.json, but I agree that it is not very clear or verbose. Happy to review a PR, go for it! 👍 

> You can specify an extra_context dictionary that will override values from cookiecutter.json or .cookiecutterrc:

**Comment by michaeljoseph** [comment]
Fixed by #864.
